velox/docs/functions/spark/array.rst

```diff
@@ -134,13 +134,34 @@ Array Functions
 .. spark:function:: array_sort(array(E)) -> array(E)
 
     Returns an array which has the sorted order of the input array(E). The elements of array(E) must
-    be orderable. Null elements will be placed at the end of the returned array. ::
+    be orderable. Null/NaN elements will be placed at the end of the returned array, with NaN elements appearing before Null elements for float types. ::
 
         SELECT array_sort(array(1, 2, 3)); -- [1, 2, 3]
         SELECT array_sort(array(3, 2, 1)); -- [1, 2, 3]
         SELECT array_sort(array(2, 1, NULL); -- [1, 2, NULL]
         SELECT array_sort(array(NULL, 1, NULL)); -- [1, NULL, NULL]
         SELECT array_sort(array(NULL, 2, 1)); -- [1, 2, NULL]
+        SELECT array_sort(array(4.0, NULL, float('nan'), 3.0)); -- [3.0, 4.0, NaN, NULL]
+        SELECT array_sort(array(array(), array(1, 3, NULL), array(NULL, 6), NULL, array(2,1))); -- [[], [NULL, 6], [1, 3, NULL], [2, 1], NULL]
+
+.. spark:function:: array_sort(array(T), function(T,U)) -> array(T)
+    :noindex:
+
+    Returns the array sorted by values computed using specified lambda in ascending order. ``U`` must be an orderable type.
+    Null/NaN elements returned by the lambda function will be placed at the end of the returned array, with NaN elements appearing before Null elements. This functions is not supported in Spark and is only used inside velox for rewring :spark:func: array_sort(array(T), function(T,T,U)) as :spark:func: array_sort(array(T), function(T,U)). ::
```

Comment:
> for rewring :spark:func: array_sort(array(T), function(T,T,U)) as :spark:func: array_sort(array(T), function(T,U))

The two references do not work well. Perhaps refer the style for array_sort on L159.